src/ln/channelmonitor.rs

```diff
@@ -1903,6 +1918,10 @@ impl ChannelMonitor {
 								// has timed out, or we screwed up. In any case, we should now
 								// resolve the source HTLC with the original sender.
 								payment_data = Some(((*source).clone(), htlc_output.payment_hash));
+							} else if let Storage::Local { ref current_remote_commitment_txid, .. } = self.key_storage {
+								check_htlc_valid_remote!(current_remote_commitment_txid, htlc_output);
```

Comment:
Can we move these checks into the if Some(...) = self.remote_claimable_outpoints.get(&input.previous_output.txid) conditional below? They don't make any sense in the spends-local-commitment-transaction branch.

Reply:
Kept it where it is but used $local_tx to avoid iterating twice on htlc_outputs